### PR TITLE
149 - Add Lookup Component and related Datagrid Features

### LINKED ISSUE #149: IdsLookup: Add new web component
**Describe the bug**

Bring over the current lookup component.  See if we can simplify and improve the code at all while improving as we go. Current lookup component https://master-enterprise.demo.design.infor.com/components/lookup

```
Markup to be added
```

**Functionality to bring over**
- Requires popup/ datagrid with selection 
- Basic states [clearable, editable, read only disabled]
- Requires pager 

**Additional context**
- 100% test coverage
- DOCS, CHANGE log ect. 
- https://github.com/infor-design/enterprise-wc/blob/master/doc/CHECKLIST.MD
- themes 

See completed components like card, tag, badge, hyperlink as reference

### PR DESCRIPTION
**Pull request by tmcconechy**
**Explain the details for making this change. What existing problem does the pull request solve?**

Adds a MVD ids-lookup. And also some related datagrid features including

- List Style Datagrid
- Selection (single/mixed/multiple)
- Also some cleanup in trigger field to make things a bit easier
- Needed fixes to modal/popup for the lookup
- Some more test coverage

**Related github/jira issue (required)**:
Fixes #149 

**Steps necessary to review your pull request (required)**:

**Test Lookup**
- go to http://localhost:4300/ids-lookup
- click the first lookup and select from the list (results reflected in the field)
- Try to test the other examples in this page
- watch console for events

**Test List Style**
- go to http://localhost:4300/ids-data-grid/list-style.html
- check list style / test themes ect

**Test Selection Modes**
- go to http://localhost:4300/ids-data-grid/single-selection.html and test selection (note that this one has supressDeselection on which means you cant unselect a row)
- watch console for events
- go to http://localhost:4300/ids-data-grid/mixed-selection.html and test mixed select (not this means clicking the row activates and clicking the checkbox selects)
- watch console for events
- go to http://localhost:4300/ids-data-grid/multiple-selection.html and test multi select
- watch console for events

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Review by jmacaluso711 — COMMENTED** [review]
Looks good.. A couple minor things I noticed and was wondering about:

The other examples on http://localhost:4300/ids-lookup are empty modals.. is that intentional?
Screen Shot 2021-11-15 at 1 45 57 PM

Looks like the [single selection](http://localhost:4300/ids-data-grid/single-selection.html) dark theme needs a bit of adjusting:
Screen Shot 2021-11-15 at 1 43 56 PM

**Review by jmacaluso711 — DISMISSED** [review]

**Review by jmacaluso711 — APPROVED** [review]

**Review by EdwardCoyle — DISMISSED** [review]
I found one minor thing but willing to approve and punt it to a new issue:

Screen Shot 2021-11-23 at 4 18 50 PM

The blue "selected" state in the background doesn't seem to be accounting for horizontal scrolling

**Comment by tmcconechy** [comment]
@EdwardCoyle fixed that was a minor styling thing and also merged main

**Review by andriivarhanov — COMMENTED** [review]
Looks good, I've added some items I've found, nothing major. Also wanted to ask if we're still doing standalone css, haven't noticed it in recent components

File: src/components/ids-lookup/ids-lookup.js [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+   */
+  set validate(value) {
+    if (value) {
+      this.setAttribute(attributes.VALIDATE, value.toString());
+      this.input.setAttribute(attributes.VALIDATE, value.toString());
+    } else {
+      this.removeAttribute(attributes.VALIDATE);
+      this.input.removeAttribute(attributes.VALIDATE);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/components/ids-lookup/ids-lookup.js:336]: Punted on this as its a problem in ids-trigger field https://github.com/infor-design/enterprise-wc/issues/443

File: src/components/ids-lookup/ids-lookup.js [hunk 2]
```diff
@@ -0,0 +1,468 @@
...
+  /**
+   * Set the string delimiter on selection
+   * @param {string} value The field name
+   */
+  set delimiter(value) {
+    if (value) {
+      this.setAttribute(attributes.DELIMITER, value);
+    }
+  }
+
```
[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/components/ids-lookup/ids-lookup.js:391]: It does work when you reselect again. Which i think is good enough for now. Added to the readme.

File: src/components/ids-lookup/README.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+- Error: The state where the user has interacted with the field and received an error. The lookup field is able to explain what caused the error below.
+
+## Keyboard Guidelines
+
+- <kbd>Tab/Shift+Tab</kbd>: Moves focus into the field to/from the next focusable item in the tab order.
+- <kbd>Down Arrow</kbd>: Opens the dialog if the input is enabled
+- <kbd>Esc</kbd>: Cancels and closes the open dialog
+- <kbd>Tab/Shift+Tab</kbd>: Tab and Shift Tab when the dialog is open, tab will move around the items, for example, from the search to the data grid
+- <kbd>Down/Up Arrow</kbd>: When focus is on the grid in the dialog this moves the focus up and down on the rows
+- <kbd>Enter/Space</kbd>: Toggle selection on the current row if multiselect. If single select, the row is selected and inserted if autoApply is enabled.
```
[the remark this answers is not in the document]
Comment (tmcconechy) [reply src/components/ids-lookup/README.md:76]: Space is correct , enter doesnt do it referencing https://main-enterprise.demo.design.infor.com/components/datagrid/example-multiselect.html so I updated the comment :)

**Comment by tmcconechy** [comment]
Thanks @andriivarhanov good comments. I addressed the majority of them. Regarding standalone CSS yes we should do this. However the ids-trigger field cannot currently do it. So i skipped it. Its probably going to be tough because of the structure/composition of that one. So we can see if its needed.

The datagrid shows a basic table however.

**Pull request merged**